src/app/api/game-sessions/route.ts

```diff
@@ -1,40 +1,371 @@
+      with: {
+        gameSessionSchedules: {
+          where: eq(gameSessionSchedules.weekday, getWeekday(gameSessionDate)),
+        },
+      },
+      where: or(
+        and(
+          lte(semesters.startDate, gameSessionDate),
+          gt(semesters.breakStart, gameSessionDate),
+        ),
+        and(
+          lt(semesters.breakEnd, gameSessionDate),
+          gte(semesters.endDate, gameSessionDate),
+        ),
+      ),
+    });
+
+    const gameSessionSchedule = semester?.gameSessionSchedules[0];
+
+    if (!gameSessionSchedule) {
+      // If no game session schedule found for the weekday within the semester, return
+      return new Response(null, { status: 204 });
+    }
+
+    const gameSessionException = await db.query.gameSessionExceptions.findFirst(
+      {
+        where: eq(gameSessionExceptions.date, gameSessionDate),
+      },
+    );
+
+    if (gameSessionException) {
+      if (gameSessionException.isDeleted) {
+        return new Response(null, { status: 204 });
+      }
+
+      return NextResponse.json(
+        {
+          date: gameSessionDate,
+          gameSessionScheduleId: null,
+          bookingOpen: null,
+          bookingClose: null,
+          startTime: gameSessionException.startTime,
+          endTime: gameSessionException.endTime,
+          locationName: gameSessionException.locationName,
+          locationAddress: gameSessionException.locationAddress,
+          capacity: gameSessionException.capacity,
+          casualCapacity: gameSessionException.casualCapacity,
+        },
+        { status: 200 },
+      );
+    }
+
+    // If schedule found and no exception, return a gameSession-like object
+    return NextResponse.json(
+      {
+        date: gameSessionDate,
+        gameSessionScheduleId: null,
+        bookingOpen: null,
+        bookingClose: null,
+        startTime: gameSessionSchedule.startTime,
+        endTime: gameSessionSchedule.endTime,
+        locationName: gameSessionSchedule.locationName,
+        locationAddress: gameSessionSchedule.locationAddress,
+        capacity: gameSessionSchedule.capacity,
+        casualCapacity: gameSessionSchedule.casualCapacity,
+      },
+      { status: 200 },
+    );
+  } catch (error) {
+    if (error instanceof z.ZodError) {
+      return NextResponse.json(error.issues, { status: 400 });
+    }
     return new Response("Internal Server Error", { status: 500 });
   }
 }
 
-/**
- * Creates a new game session
- */
 export async function POST(req: NextRequest) {
   try {
-    const json = await req.json();
+    const user = await getCurrentUser();
+    if (!user) {
+      return new Response("ERROR: Unauthorized request", { status: 401 });
+    }
+    if (user.role != "admin") {
+      return new Response("ERROR: No valid permissions", { status: 403 });
+    }
 
-    const body = insertGameSessionSchema.parse(json);
+    const gameSessionToInsert = insertGameSessionSchema.parse(await req.json());
```

Comment:
No longer have bookingOpen & bookingClose fields -- you might have to make a new schema variation for insertGameSessionExceptions where those fields are omitted but everything else is mandatory?